### PR TITLE
WIP support searchAround* methods on an ObjectSet

### PR DESCRIPTION
**Pull request by mfedderly**

### CODE REVIEW

File: packages/legacy-client/src/client/objectSets/OsdkObjectSet.ts [hunk 1]
```diff
@@ -117,7 +126,7 @@ export function createBaseOsdkObjectSet<
   };
 
   const objectSet: BaseObjectSetOperations<OsdkLegacyObjectFrom<O, K>> = {
-    apiName: apiName as OsdkLegacyObjectFrom<O, K>["__apiName"],
+    apiName: apiName as string as OsdkLegacyObjectFrom<O, K>["__apiName"],
```
Comment (mfedderly) [inline packages/legacy-client/src/client/objectSets/OsdkObjectSet.ts:129]: This cast was sad, but I think you can make this a `K`?

File: packages/legacy-client/src/client/objectSets/createObjectSetSearchAround.ts [hunk 2]
```diff
@@ -15,12 +15,51 @@
...
+      } satisfies SearchAroundObjectSetDefinition;
+
+      const objSet = createOsdkObjectSet(
+        clientContext,
+        targetType,
+        definition,
+        ontologyDefinition,
+      );
+
+      return objSet as ObjectSet<O, any>;
```
Comment (mfedderly) [inline packages/legacy-client/src/client/objectSets/createObjectSetSearchAround.ts:56]: For some reason this isn't satisfying the type on `result[key]` and I haven't figured it out yet

File: packages/legacy-client/src/client/interfaces/searchAround.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ type InferLinkType<T> = T extends SingleLink<infer V> ? V
 export type SearchAround<T extends OntologyObject> = {
   [
     K in Extract<keyof T, string> as IsLink<T[K]> extends true
-      ? `searchAround${Capitalize<string & K>}`
+      ? `searchAround${Capitalize<K>}`
```
Comment (mfedderly) [inline packages/legacy-client/src/client/interfaces/searchAround.ts:32]: Extract<> above already enforces that K is a string

File: packages/legacy-client/src/client/objectSets/OsdkObjectSet.test.ts [hunk 4]
```diff
@@ -53,4 +53,32 @@ describe("OsdkObjectSet", () => {
...
+      baseObjectSet,
+      MockOntology,
+    );
+
+    const searchAroundObjectSet = os.searchAroundTask();
+
+    expect(searchAroundObjectSet.definition).toEqual({
+      type: "searchAround",
+      objectSet: baseObjectSet,
+      link: "Task",
```
Comment (mfedderly) [inline packages/legacy-client/src/client/objectSets/OsdkObjectSet.test.ts:81]: capital T in task is correct, right?

Comment (pashakondratyev) [reply packages/legacy-client/src/client/objectSets/OsdkObjectSet.test.ts:81]: In this case yeah, It should be whatever the ApiName from the definition is

**Review by pashakondratyev — APPROVED** [review]

**Pull request merged**